### PR TITLE
Adds layer index verification routines and Storm API plumbing

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: synapse/lib/layer.py [hunk 1]
```diff
@@ -1204,6 +1213,182 @@ async def __anit__(self, layrinfo, dirn, nexsroot=None, allow_upstream=True, map
 
         self.onfini(self._onLayrFini)
 
+    async def verifyBuidTag(self, buid, formname, tagname, tagvalu):
+        abrv = self.tagabrv.bytsToAbrv(tagname.encode())
+        abrv += self.getPropAbrv(formname, None)
+        if not self.layrslab.hasdup(abrv, buid, db=self.bytag):
+            yield ('NoTagIndex', {'buid': buid, 'form': formname, 'tag': tagname, 'valu': tagvalu})
+
+    def _testDelTagIndx(self, buid, form, tag):
```
Comment (vEpiphyte) [inline synapse/lib/layer.py:1222]: Should these test methods call layr sync so we ensure they are commited txns?

File: synapse/lib/layer.py [hunk 2]
```diff
@@ -1204,6 +1213,182 @@ async def __anit__(self, layrinfo, dirn, nexsroot=None, allow_upstream=True, map
...
+            await asyncio.sleep(0)
+
+            sode = self._getStorNode(buid)
+            if sode is None:
+                yield ('NoBuidForTagIndex', {'buid': buid, 'tag': tag})
+                continue
+
+            tags = sode.get('tags')
+            if tags is None:
+                yield ('NoTagForTagIndex', {'buid': buid, 'tag': tag})
```
Comment (vEpiphyte) [inline synapse/lib/layer.py:1294]: This error differs from the next one an represent different conditions?

File: synapse/lib/layer.py (hunk 1 above)
Comment (invisig0th) [reply synapse/lib/layer.py:1222]: sure :)

File: synapse/lib/layer.py (hunk 2 above)
Comment (invisig0th) [reply synapse/lib/layer.py:1294]: Not really, they both represent the non-presence of a tag value it's just that we need 2 cases to account for their being ( or not being ) any others.

File: synapse/lib/layer.py (hunk 1 above)
Comment (invisig0th) [reply synapse/lib/layer.py:1222]: Actually, if you mean commit() it's def not necessary

**Review by vEpiphyte — APPROVED** [review]

**Pull request merged**